File: app/src/main/java/com/hywy/luanhzt/receiver/RestartAppReceiver.java
package com.hywy.luanhzt.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * 杀死app进程并重启
 * Created by Superman on 2018/7/26.
 */

public class RestartAppReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals("restart.app")) {
            Log.e("TAG", "接受到广播");
            Intent i = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            context.startActivity(i);
        }
    }
}

File: app/src/main/java/com/hywy/luanhzt/receiver/RestartAppReceiver.kt
package com.hywy.luanhzt.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * 杀死app进程并重启
 * Created by Superman on 2018/7/26.
 */

class RestartAppReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == "restart.app") {
            Log.e("TAG", "接受到广播")
            val i = context.packageManager.getLaunchIntentForPackage(context.packageName)
            i!!.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            context.startActivity(i)
        }
    }
}
